src/main/java/com/exasol/releasedroid/main/ReleaseDroid.java

```diff
@@ -51,21 +51,28 @@ public ReleaseDroid(final RepositoryGateway repositoryGateway, final UseCase val
     public void run(final UserInput userInput) {
         validateUserInput(userInput);
         final Repository repository = this.repositoryGateway.getRepository(userInput);
-
         final ReleasePlatforms platforms = ReleasePlatforms.from(userInput, repository);
         LOGGER.fine(() -> "Release Droid has received '" + userInput.getGoal() + "' request for the project '"
                 + userInput.getFullRepositoryName() + "'.");
         final List<Report> reports = new ArrayList<>();
-        final boolean isValidate = userInput.getGoal() != Goal.RELEASE;
-        final UseCase useCase = isValidate ? this.validateUseCase : this.releaseUseCase;
+
         final Optional<Path> releaseGuide = userInput.releaseGuide();
-        if (releaseGuide.isPresent() && isValidate) {
+        if (releaseGuide.isPresent()) {
             ReleaseGuide.from(repository).write(releaseGuide.get());
         }
-        reports.addAll(useCase.apply(repository, platforms));
+        // always apply use case "validate" first
```

Comment:
This would belong into the design document, not into the code. In the code it is clear  that this always comes first by reading the code.

Reply:
👍

Reply:
Funny: Actually there was a design already `dsn~rd-starts-release-only-if-all-validation-succeed~1` but from my understanding the needed impl it was covered in the wrong place.